### DISCUSSION TITLE
LDAP IDP Authenticates All External Users as Same User

### DISCUSSION THREAD

**Discussion by MicahMoore** [zitadel/zitadel#8277] [Q&A] [answered]
Hello!

**TLDR: Regardless of which external user I sign in as, it always logs me in as the first user.**

I am using Active Directory as my IDP. Here are my settings:

<img width="721" alt="image" src="https://github.com/zitadel/zitadel/assets/20447381/4bb712d5-38ee-467b-b944-3b23ddd16a5e">

<img width="721" alt="image" src="https://github.com/zitadel/zitadel/assets/20447381/53035613-f56c-4d13-bca4-50416e4c55d7">

If I have no external users and I login for the first time I receive this page:

<img width="721" alt="Screenshot 2024-07-09 at 9 37 45 PM" src="https://github.com/zitadel/zitadel/assets/20447381/b06d3d8a-0075-417c-8ffa-02658bc4b6eb">

Once I enter my information I am able to signin and the account is created.

<img width="987" alt="Screenshot 2024-07-09 at 9 41 44 PM" src="https://github.com/zitadel/zitadel/assets/20447381/4c93aa2f-d841-4f43-83e5-896f48ca96c9">

The problem is if I sign in with a different external account, different username, different password, It still signs me in as the first test account.

<img width="987" alt="Screenshot 2024-07-09 at 9 43 30 PM" src="https://github.com/zitadel/zitadel/assets/20447381/9d8f1c23-7729-4ddc-a296-0f38e5423bcf">

<img width="347" alt="Screenshot 2024-07-09 at 9 44 34 PM" src="https://github.com/zitadel/zitadel/assets/20447381/0dc14dc8-c1ed-49ab-9aa7-3a9bb96f3289">

Any guidance would be appreciated!

**Comment by hifabienne** [comment]
@stebenz can you help here?

**Comment by stebenz** [comment]
Hi @MicahMoore 
As you configure 'userprincipalname' as ID attribute, do the different users on your LDAP have different values in this attribute?
The ID is used to find out which user is linked in ZITADEL.

**Reply by MicahMoore** [reply]
Hi @stebenz

userprincipalname is username@domain, so the values should be different for both users. Is there a different attribute you suggest I should try out?

**Reply by MicahMoore** [reply, marked as answer]
I figured it out! A case of the case sensitivity. Once i put "userPrincipalName" everything worked flawlessly.

I was hoping I could tack on a second question here. What does the Email Verified Attribute do? I put "true" assuming it would bypass the email verification for new users but that was not the case.

![image](https://github.com/zitadel/zitadel/assets/20447381/2925c025-fa32-407b-9638-22fdc2a62211)

**Reply by stebenz** [reply]
Ah perfect, case-sensetivity.
No all the attributes in the IDP configuration are only for mapping, would mean that if you provide a boolean parsable attribute from LDAP which can tell if the email is verified, that would have your desired effect.

**Reply by MicahMoore** [reply]
Got it, thanks!
